lib/openstack_query/query_blocks/query_builder.py

```diff
@@ -66,6 +70,23 @@ def server_side_filters(self, server_filters: ServerSideFilters):
         """
         self._server_side_filters = server_filters
 
+    @property
+    def server_filter_fallback(self):
+        """
+        a getter method to return equivalent client-side filters for each server-side filter if
+        server-side filters are not to be used
+        """
+        return self._server_filter_fallback
+
+    @server_filter_fallback.setter
+    def server_filter_fallback(self, fallback_filters: ClientSideFilters):
+        """
+        a setter method to set client-side filters to fallback on for each server-side filter if
+        server-side filters are not to be used
+        :param fallback_filters: a set of client-side filters
+        """
+        self._server_filter_fallback = fallback_filters
+
     def parse_where(
```

Comment:
Can we drop this and simply append server-side filters to the client-side list on subset?
That way we still only have two modes (client / server) for filtering, since the effect rather than why we have it is the important part?

Reply:
we only register subsets at time of running - hence we need to hold a separate list here in query builder